openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -115,61 +115,53 @@ def _build_jwt_response_from_access_token_response(self, request, response):
         client_id = self._get_client_id(request)
         adapter = self.get_adapter(request)
-        expires_in, scopes, user = self._decompose_access_token_response(adapter, response)
-        issuer, secret, audience, filters, is_client_restricted = self._get_client_specific_claims(
-            client_id,
-            adapter
-        )
+        is_client_restricted = adapter.is_client_restricted(client_id)
+
+        expires_in, scope, user = self._parse_access_token_response(adapter, response)
+        jwt_builder = self._get_jwt_builder(user, is_client_restricted)
+
         content = {
-            'access_token': JwtBuilder(
-                user,
-                secret=secret,
-                issuer=issuer,
-            ).build_token(
-                scopes,
+            'access_token': jwt_builder.build_token(
+                scope.split(' '),
                 expires_in,
-                aud=audience,
                 additional_claims={
-                    'filters': filters,
+                    'filters': adapter.get_authorization_filters(client_id),
                     'is_restricted': is_client_restricted,
                 },
             ),
             'expires_in': expires_in,
+            'scope': scope,
             'token_type': 'JWT',
-            'scope': ' '.join(scopes),
         }
         return json.dumps(content)
 
-    def _decompose_access_token_response(self, adapter, response):
-        """ Decomposes the access token in the request to an expiration date, scopes, and User. """
+    def _parse_access_token_response(self, adapter, response):
+        """ Parses the expires_in, scope, and user values of the response. """
         content = json.loads(response.content)
         access_token = content['access_token']
+        expires_in = content['expires_in']
         scope = content['scope']
-        scopes = scope.split(' ')
         user = adapter.get_access_token(access_token).user
-        expires_in = content['expires_in']
-        return expires_in, scopes, user
+        return expires_in, scope, user
 
-    def _get_client_specific_claims(self, client_id, adapter):
-        """ Get claims that are specific to the client. """
+    def _get_jwt_builder(self, user, is_client_restricted):
         # If JWT scope enforcement is enabled, we need to sign tokens
-        # given to restricted application with a separate secret which
+        # given to restricted application with a separate key which
         # other IDAs do not have access to. This prevents restricted
         # applications from getting access to API endpoints available
         # on other IDAs which have not yet been protected with the
         # scope-related DRF permission classes. Once all endpoints have
-        # been protected we can remove this if/else and go back to using
-        # a single secret.
+        # been protected we can remove this check and enable all IDAs
```

Comment:
Is "this check", a check for restricted applications?  Can we not use "this"?

Reply:
Updated wording.